laws/src/main/scala/org/http4s/laws/discipline/arbitrary.scala

```diff
@@ -731,14 +731,17 @@ private[discipline] trait ArbitraryInstances { this: ArbitraryInstancesBinCompat
   implicit val http4sTestingCogenForUserInfo: Cogen[Uri.UserInfo] =
     Cogen.tuple2[String, Option[String]].contramap(u => (u.username, u.password))
 
+  def genMaybePort: Gen[Option[Int]] =
+    Gen.option(posNum[Int].suchThat(port => port >= 0 && port <= 65536))
```

Comment:
0 and 65536 are not valid ports. 0 has a special meaning of "please choose a port for me". 65536 doesn't fit in two bytes.

There is more efficient way to generate a number in chosen range than via `suchThat`. In this case it could be
```
    Gen.option(chooseNum(1, 65535))
```